Tidb Controller manager crash.

## Bug Report

**What did you expect to see?**
```
E0323 21:40:15.809912       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 550 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1afe7a0?, 0xc001806180})
        k8s.io/apimachinery@v0.19.16/pkg/util/runtime/runtime.go:74 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000c42ba0?})
        k8s.io/apimachinery@v0.19.16/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1afe7a0, 0xc001806180})
        runtime/panic.go:838 +0x207
github.com/pingcap/tidb-operator/pkg/manager/member.(*tidbMemberManager).syncInitializer(0xc0008a2dc0, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/manager/member/tidb_member_manager.go:270 +0x1191
github.com/pingcap/tidb-operator/pkg/manager/member.(*tidbMemberManager).Sync(0xc000a04520?, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/manager/member/tidb_member_manager.go:117 +0x26c
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).updateTidbCluster(0xc0008d4600, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_control.go:218 +0x214
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).UpdateTidbCluster(0xc0008d4600, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_control.go:107 +0xb0
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).syncTidbCluster(...)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:159
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).sync(0xc0008b2ed0, {0xc00130d220, 0xd})
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:155 +0x278
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).processNextWorkItem(0xc0008b2ed0)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:122 +0x108
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).worker(0xc0008366a0?)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:110 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x1c0b1df?)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x1?, {0x1f435a0, 0xc0014942a0}, 0x1, 0xc000a3dbc0)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0xd5?, 0xc0007372a0?)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x1b44860?, 0xc000735f20?, 0xc00076d760?)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:90 +0x25
created by github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).Run
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:102 +0x157
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 550 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000c42ba0?})
        k8s.io/apimachinery@v0.19.16/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1afe7a0, 0xc001806180})
        runtime/panic.go:838 +0x207
github.com/pingcap/tidb-operator/pkg/manager/member.(*tidbMemberManager).syncInitializer(0xc0008a2dc0, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/manager/member/tidb_member_manager.go:270 +0x1191
github.com/pingcap/tidb-operator/pkg/manager/member.(*tidbMemberManager).Sync(0xc000a04520?, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/manager/member/tidb_member_manager.go:117 +0x26c
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).updateTidbCluster(0xc0008d4600, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_control.go:218 +0x214
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).UpdateTidbCluster(0xc0008d4600, 0xc000be7b80)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_control.go:107 +0xb0
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).syncTidbCluster(...)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:159
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).sync(0xc0008b2ed0, {0xc00130d220, 0xd})
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:155 +0x278
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).processNextWorkItem(0xc0008b2ed0)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:122 +0x108
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).worker(0xc0008366a0?)
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:110 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x1c0b1df?)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x1?, {0x1f435a0, 0xc0014942a0}, 0x1, 0xc000a3dbc0)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0xd5?, 0xc0007372a0?)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x1b44860?, 0xc000735f20?, 0xc00076d760?)
        k8s.io/apimachinery@v0.19.16/pkg/util/wait/wait.go:90 +0x25
created by github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).Run
        github.com/pingcap/tidb-operator/pkg/controller/tidbcluster/tidb_cluster_controller.go:102 +0x157

```
**What did you see instead?**

--- a/pkg/manager/member/tidb_member_manager.go
+++ b/pkg/manager/member/tidb_member_manager.go
@@ -267,7 +267,7 @@ func (m *tidbMemberManager) syncInitializer(tc *v1alpha1.TidbCluster) {
 		return
 	}
 	// TiDB service has endpoints
-	if eps != nil && len(eps.Subsets[0].Addresses) > 0 {
+	if eps != nil && len(eps.Subsets) != 0 && len(eps.Subsets[0].Addresses) > 0 {
 		isTiDBReady = true
 	}
 

--- a/pkg/manager/member/tidb_member_manager.go
+++ b/pkg/manager/member/tidb_member_manager.go
@@ -267,7 +267,7 @@ func (m *tidbMemberManager) syncInitializer(tc *v1alpha1.TidbCluster) {
 		return
 	}
 	// TiDB service has endpoints
-	if eps != nil && len(eps.Subsets) != 0 && len(eps.Subsets[0].Addresses) > 0 {
+	if eps != nil && len(eps.Subsets) > 0 && len(eps.Subsets[0].Addresses) > 0 {
 		isTiDBReady = true
 	}
 

